### PR TITLE
feat: update quantity calculated documents of AppBio Quantstudio

### PR DESCRIPTION
**Pull request by alejandro-salgado**
- update quantity calculated documents of AppBio Quantstudio by adding Y-intercept and Slope as data sources

### CODE REVIEW

File: src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_calculated_documents.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ def build_quantity(well_item: WellItem) -> CalculatedDocument | None:
         value=quantity,
         data_sources=[
             DataSource(feature="cycle threshold result", reference=well_item),
+            DataSource(feature="Y-intercept", reference=well_item),
+            DataSource(feature="Slope", reference=well_item),
```
Comment (mdlu) [inline src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_calculated_documents.py:34]: question: seems like the feature names aren't capitalized elsewhere, should it be `y-intercept` and `slope` throughout this PR, to match?

Comment (alejandro-salgado) [reply src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_calculated_documents.py]: yes, you are right. I change them

File: tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_multiple_cal_doc_wells.json [hunk 2]
```diff
@@ -495,6 +496,14 @@
                             {
                                 "data source identifier": "APPBIO_QUANTSTUDIO_TEST_ID_0",
                                 "data source feature": "cycle threshold result"
+                            },
```
Comment (nathan-stender) [inline tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_multiple_cal_doc_wells.json:499]: How are we "referencing" y-intercept and slope when they do not appear in the measurements? 

I see in the test files that there are columns for `y-intercept` and `slope` - shouldn't we add them to the processed data for this to make sense?

**Review by nathan-stender — CHANGES_REQUESTED** [review]

File: tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_multiple_cal_doc_wells.json (hunk 2 above)
Comment (alejandro-salgado) [reply tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_multiple_cal_doc_wells.json:499]: Added check to add data sources only when their values are non None.

**Review by mdlu — APPROVED** [review]

File: tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_multiple_cal_doc_wells.json (hunk 2 above)
Comment (nathan-stender) [reply tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_multiple_cal_doc_wells.json:499]: @alejandro-salgado I'm still not clear that this is addressed. My point is not that the measurement could be None, my point is that the `y-intercept` and `slope` do not appear in the data source. There is no entry for `y-intercept` and `slope` in the measurement's processed data. I think you need to add them to the processed data?

**Pull request merged**